lib/promoted/ruby/client.rb

```diff
@@ -57,8 +57,16 @@ def initialize (params={})
           @metrics_timeout_millis  = params[:metrics_timeout_millis] || DEFAULT_METRICS_TIMEOUT_MILLIS
 
           @http_client = FaradayHTTPClient.new
-          @pool = Concurrent::CachedThreadPool.new
           @validator = Promoted::Ruby::Client::Validator.new
+
+          # Thread pool to process delivery of shadow traffic. Will silently drop exccess requests beyond the queue
+          # size, and silently eat errors on the background threads.
+          @pool = Concurrent::ThreadPoolExecutor.new(
+            min_threads: 0,
+            max_threads: 10,
```

Comment:
I think this is fine for now.  Worst-case, if we do a 1s timeout on Delivery API calls, this can handle ~10 qps.  Hopefully latency is faster.

Reply:
Note that I also set max_queue, which will mitigate bursts.